### ISSUE TITLE
Way for storing data in user home folder

### ISSUE THREAD

**Issue by DenSpirit** [GoldenGnu/jeveassets#63]
Currently jEveAssets can update it's data and itself if there is an update. On most Linux systems programs (as packages) are managed by package manager (pacman in Archlinux's case).

To update itself on Linux, program would need write access to root folder, which is usually granted by running it with `sudo`. However, this creates a blank spot in package manager's understanding of whereabouts of package data and is considered a security risk. Some users are ok with that, but an alternative location would be nice.

**Comment by GoldenGnu** [comment]
The only way I can thing of to make this possible (but, still optional) is to add a file to program directory to configure the behavior.

I expect you want an option to disable the internal auto update?
How would you prefer the static data location settings worked: just a boolean to enable using the user directory or would you rather be able to set the exact path?

**Comment by DenSpirit** [comment]
Regarding disabling auto update, I think jEveAssets already has the `-noupdate` option.

On Linux, it's common to use a specific directory in user folder, like `~/.cache/<program>` or `~/.program`, and populate it on first start of the app.

**Comment by GoldenGnu** [comment]
Regarding the -noupdate option I actually think that is meant to block all updates for prices/esi/evekit and not from date/program updates. I don't know if it still works to be honest, as the update core was completely rewritten in when I ported to ESI and I only use the command line options when debugging and I haven't had the need to use -noupdate for ages.

jEveAssets should already store all dynamic data in ~/.jeveassets and storing the static data there is merely a question of find a way to let jEveAssets know to look for the data there instead of in the program directory. Can you specify command line options when creating packages? because, that would be really simple to implement something like -package, that change everything you would want changed.

**Comment by DenSpirit** [comment]
In Arch distribution I have a .desktop file and .sh file to launch jEveAssets and I can start the app with any options.

**Comment by GoldenGnu** [comment]
okay, we're rolling with -package

So, that will disable auto update and look for the static data in ~/.jeveassets/data/
I will try to get it done for next release.
I will post a dev build here before release, so you can give feedback.

**Comment by DenSpirit** [comment]
Can jEveAssets will still try to auto-update, but save the data in user folder?

**Comment by GoldenGnu** [comment]
Yes, that would be possible.

I will have to update jUpdate project, so, it will start jEveAssets in package mode, when needed, but, that shouldn't be too much of a problem.

**Comment by GoldenGnu** [comment]
So, I had a break from jEveAssets in October and I accidentally forgot about this issue.
But, I'm going to start working on it now.

If possible, could you join twitterfleet slack (https://www.fuzzwork.co.uk/tweetfleet-slack-invites/) and message me @goldengnu or join the jEveAssets discord server (https://discordapp.com/invite/8kYZvbM). (whatever you prefer).

**Comment by GoldenGnu** [comment]
I have a few questions:
1) Would it be possible to include  an extra file to identify who are managing the package and enabling the package mode - the file would be in the same folder as the jar file?
2) Would it be possible to update the package when the data is update? It turns out changing where the static data is download, is really really complicated, sadly.

**Comment by GoldenGnu** [comment]
Okay, this feature is mostly completed. I would still like to hear from you.
You will need to add the `packagemanager.properties` to the same directory as jeveassets.jar
[jeveassets.zip](https://github.com/GoldenGnu/jeveassets/files/3193926/jeveassets.zip)
I also included two images files, I think would be better as icons, than the one you're using now (yours looks great, but, it's not the application icon used in jEveAssets).

**Comment by GoldenGnu** [comment]
This is now completed and will be released in the next feature release.
I changed the packagemanager.properties variable to `maintainers` (instead of `name`) as it makes more sense. See, the example:
https://github.com/GoldenGnu/jeveassets/blob/feature-branch/tools/packagemanager.properties

**Comment by GoldenGnu** [comment]
From Discord (Debi Apol):
>I'd like some information about existence of new data in the app to be available still, because I may not be the maintainer for life for the package, and I may not always be available. AUR packages are not distributed in binary form but rather built from scripts on user machine. This way user can customize the package and update it locally if maintainer is slow.

So, we still need to notify the user of new updates, but, do not allow them to update. That should be done by the package manager or alternatively allowing the users to manually update.

**Comment by DenSpirit** [comment]
Notification could be done from application side here by changing the update window if a setting is enabled in packagemanager.properties.

**Comment by GoldenGnu** [comment]
that is a good idea, I will add that

**Comment by GoldenGnu** [comment]
Okay, so, I will commit this later today:

new packagemanager.properties option:
```ini
#notify users of new updates on startup  [values: true|false]
notifyUpdates=true
```

- If `true`, the user will be shown a dialog notifying them of new updates on startup (but the will not be able to update with jEveAssets auto update).
- if `false` jEveAssets will never check for new updates
- If user want to enable the auto update included injEveAssets, they can delete the `packagemanager.properties` file. That will make jEveAssets work like a normal install.

I hope that is flexible enough?
Thank you for working trough this with me.

**Comment by DenSpirit** [comment]
I think it's looking good. I'll try to package the dev build today.

**Comment by GoldenGnu** [comment]
I don't have a dev build up with those changes, I will make you one now and post in on discord

**Comment by GoldenGnu** [comment]
Released in 6.1.0
